### ISSUE TITLE
Memory leak in fit() with callback

### ISSUE THREAD

**Issue by RavindhranSankar** [tensorflow/tensorflow#34876]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>
** Coding Language **
Python

**System information**

- OS Platform and Distribution: Ubuntu 16.04
- TensorFlow installed from (source or binary): pip3
- TensorFlow version : 2.0.0
- TensorFlow Keras version: '2.2.4-tf'
- Python version: 3.6
- GPU model and memory:

![image](https://user-images.githubusercontent.com/7545134/70257890-11129e00-1740-11ea-944b-291d5fed4c30.png)

**Describe the current behavior**
CPU Ram progressively gets consumed as the fit() works. Each EPOCH consumes more and more memory. This memory leak only happens when a callback is assigned, any callback eg: tensorboard.

The memory consumption can be seen in both ways:
-  System memory (vmstat)
-  Printing resource memory usage in code (print(resource.getrusage(resource.RUSAGE_SELF).ru_maxrss))

Attached the vmstat print to screen:
![image](https://user-images.githubusercontent.com/7545134/70258980-196bd880-1742-11ea-9cfe-6d6fa684ac3b.png)

**Code to reproduce the issue**
BATCH_SIZE = 24
learning_rate = 0.000005
decay_rate = 1e-6
EPOCHS = 4000
CLIP_VALUE = 1.5
l1reg = 0.01
l2reg = 0.00

model = Sequential()

model.add(LSTM(512, input_shape=(train_x.shape[1], train_x.shape[2]), activation='relu',
               return_sequences=True, bias_regularizer=L1L2(l1=l1reg, l2=l2reg)))
model.add(Dropout(0.3))
model.add(BatchNormalization())

model.add(LSTM(512, activation='relu', return_sequences = True))
model.add(Dropout(0.3))
model.add(BatchNormalization())

model.add(LSTM(256, activation='relu'))
model.add(Dropout(0.25))
model.add(BatchNormalization())

model.add(Flatten())
model.add(Dense(24, activation='relu'))

model.add(Flatten())
model.add(Dense(2, activation='softmax'))
opt = tf.keras.optimizers.Adam(lr=learning_rate, decay=decay_rate, clipvalue=CLIP_VALUE)

model.compile(
    loss='sparse_categorical_crossentropy',
    optimizer=opt,
    metrics=['accuracy'],
)

NAME = f"PARAM-{train_x.shape[2]}-FIRSTTEST-{int(time.time())}"
print(NAME)
print('-----------------------------------------')
tensorboard = TensorBoard(log_dir=f"{ML_LOG_DIR}/{NAME}")
filepath = "RNN_Final-{epoch:02d}-{val_accuracy:.3f}"
checkpoint = ModelCheckpoint("{}/{}.model".format(ML_MODEL_DIR, filepath, monitor='val_accuracy',
                            verbose=1, save_best_only=True, mode='max')) # saves only the best ones

history = model.fit(
        train_x,
        train_y,
        epochs=EPOCHS,
        batch_size=BATCH_SIZE,
        validation_data=(test_x, test_y)
        callbacks=[tensorboard, checkpoint]
)

**Other info / logs**
The problem goes away if the callbacks are removed.

**Comment by oanush** [comment]
@RavindhranSankar ,
Can you please provide the complete code to reproduce issue from our end?Thanks!

**Comment by oanush** [comment]
@RavindhranSankar ,
Any update on the issue?Thanks!

**Comment by RavindhranSankar** [comment]
Hi ,

Sorry about the delay.

I attached all the code in the bug report. The rest of the code pertains to
the data and cleaning it.

I will send across the full code tomorrow

Happy Holidays

Ravi

**Comment by oanush** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by dd1923** [comment]
Please reopen this. I also ran into this issue. After spending a lot of time trying to debug things, it looks like this memory leak happens when TensorBoard callback is used. The moment I remove the Tensorboard callback, it goes away.

**Comment by dd1923** [comment]
And it seems to only happen during validation. Training seem to go fine for me. It was really difficult to find out the source in Tensorboard callback.

**Comment by RavindhranSankar** [comment]
Agreed. Same experience for me. Removing the callback makes the leak go away. However as soon as the Tensorboard callback is added, it returns. 

If I recollect correctly, the leak happens with any callback (not just Tensorboard)

**Comment by dd1923** [comment]
I tried with other callbacks of mine and also the callbacks provided by tensorflow. Are you sure it happens when you remove tensorboard? For me, it seems to work with other callbacks as long as tensorboard isn't there.

**Comment by RavindhranSankar** [comment]
@Regarding other callbacks: Let me get back on that. My memory could be misleading me. Will test again and report back

**Comment by dd1923** [comment]
Anyone actively working on this issue? Its a bug that hits a core functionality such as tensorboard

**Comment by EgOrlukha** [comment]
> Anyone actively working on this issue? Its a bug that hits a core functionality such as tensorboard

You can reduce weight of your Tensorboard callback by setting write_graph flag to False.
Source: https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard

In my case this helped, I just didn't need to analyze the full model graph.
